src/window/gui/resource/Font.cpp

```diff
@@ -5,9 +5,7 @@ Font::Font() : Resource(std::shared_ptr<ResourceInitData>()) {
 }
 
 Font::~Font() {
-    if (Data != nullptr) {
-        delete[] Data;
-    }
+    // let ImGui free the font data
```

Comment:
Not sure I love the idea of a factory allocating memory for a resource, then the resource individually not being responsible for that memory. That's the pattern in all other cases.

Can you convince me, or maybe call the imgui free functions directly?

Reply:
the imgui free are just free wrap in a function, and while the font are in imgui, it's imgui who try to free it, so we can free in the class else it will be make a double free, and don't really know how to remove a font in imgui.